clients/rospy/src/rospy/impl/tcpros_base.py

```diff
@@ -798,22 +798,11 @@ def receive_loop(self, msgs_callback):
                         msgs = self.receive_once()
```

Comment:
Does anything else rely on `self.socket` being `None`, or is it just used to manage the reconnect here?

Reply:
So, it's mostly just used at the connect stage and here at the reconnect. By calling `close()` we effectively shut down the thread and remove this connection so in theory nothing will be running anymore that checks it.
https://github.com/locusrobotics/ros_comm/blob/3971dd3b99777444f99f102009330a15cafde30c/clients/rospy/src/rospy/impl/transport.py#L102

Once `self.done` is set, the thread stops here https://github.com/locusrobotics/ros_comm/blob/3971dd3b99777444f99f102009330a15cafde30c/clients/rospy/src/rospy/impl/tcpros_pubsub.py#L424